### PR TITLE
Refactor architecture + Add redux-sagas and api

### PR DESCRIPTION
**Pull request by drewandrew**
This PR...
* Refactors the folder structure roughly following the "[Ducks](https://github.com/erikras/ducks-modular-redux)" modular approach to redux organization.
* Fills out the src/services folder with all of the Api stuff that will be hooked up in the following PR (just making the endpoints for campaigns now).
* Creates reduxFetch wrapper for fetch Api that will set the token in the headers on each request.
* Nests the creatures state under campaign to prepare to support multiple campaigns

### CODE REVIEW

**Comment by joshleecreates** [comment]
Needs to be rebased and squashed a bit

File: src/App.jsx [hunk 1]
```diff
@@ -1,26 +1,48 @@
...
 
 import Initiative from './components/Initiative';
-// import StateManagement from './components/StateManagement';
 import CreatureDetails from './components/CreatureDetails';
 import InitiativeButtonsWidget from './components/Initiative/InitiativeButtons';
+import { userSelector } from './services/api/auth/selectors';
 import './styles/index.scss';
 
 class App extends Component {
+  componentDidMount() {
```
Comment (joshleecreates) [inline src/App.jsx:11]: Empty function?

File: src/App.jsx [hunk 2]
```diff
@@ -1,26 +1,48 @@
...
-          {/*<StateManagement />*/}
-          <Initiative />
-          <div className="temporary-right-side">
-            <InitiativeButtonsWidget />
-            <CreatureDetails />
-          </div>
-        </main>
+        {user.token && user.token === document.cookie ? (
+          <main className="main">
+            <Initiative />
```
Comment (joshleecreates) [inline src/App.jsx:21]: These should probably be passed into a presentational component as children.

File: src/App.jsx [hunk 3]
```diff
@@ -1,26 +1,48 @@
...
       <div>
-        <main className="main">
-          {/*<StateManagement />*/}
-          <Initiative />
-          <div className="temporary-right-side">
-            <InitiativeButtonsWidget />
-            <CreatureDetails />
-          </div>
-        </main>
+        {user.token && user.token === document.cookie ? (
```
Comment (joshleecreates) [inline src/App.jsx:19]: this logic shouldn't go in a component. maybe a reducer?

Comment (joshleecreates) [reply src/App.jsx:19]: (or at least not inline with the JSX)

**Comment by drewandrew** [comment]
I've been using the Github "squash and merge" button which you can set as the default for this repo (maybe even organization)... also you should totally make me an admin of this repo ;)

**Review by joshleecreates — COMMENTED** [review]
Didn't review everything but a few thoughts...

**Comment by drewandrew** [comment]
@joshleecreates Yeah, I can clean this up a bit now that we're doing code reviews. I just reached a workable stopping place and created the PR so you could check it out. Plenty to fix in there still :grin:

**Pull request closed without merging**